classes/class-wc-rest-connect-migration-flag-controller.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( class_exists( 'WC_REST_Connect_Migration_Flag_Controller' ) ) {
+	return;
+}
+
+class WC_REST_Connect_Migration_Flag_Controller extends WC_REST_Connect_Base_Controller {
+	protected $rest_base = 'connect/migration-flag';
+
+	private function is_valid_state($state) {
+		$valid_states = [
+			WC_Connect_API_Constants::MIGRATION_STATE_NOT_STARTED,
+			WC_Connect_API_Constants::MIGRATION_STATE_STARTED,
+			WC_Connect_API_Constants::MIGRATION_STATE_COMPLETED,
+			WC_Connect_API_Constants::MIGRATION_STATE_FAILED
+		];
+		if (in_array($state, $valid_states, true)) {
+			return true;
+		}
+		return false;
+	}
+
+	public function post($request) {
+		$params = $request->get_json_params();
+		$migration_state = intval($params['migration_state']);
+		if (!$this->is_valid_state($migration_state)) {
+			$error = new WP_Error(
+				'invalid_migration_state',
+				__( 'Invalid migration state. Can not update migration state.', 'woocommerce-services' ),
+				array( 'status' => 400 )
+			);
+			return $error;
+		}
+
+		$result = WC_Connect_Options::update_option( 'wcshipping_migration_state', $migration_state );
```

Comment:
This is a good callout, I thought about using its own options name with the WP `update_option` directly, but later opt-ed for the `WC_Connect_Options::update_option()`. 

I leaned towards using the `WC_Connect_Options::update_option()` because it follows the existing pattern to encapsulate all connect-server-related configs under 1 option name. 

> Is the option something that will be used by WCShip?

I am not sure but I don't think so. During the migration, this flag will be updated a few times and I don't think the migration script will sync this very well either. It won't be reliable even if we migrated this option over.

Maybe it is better to use its own variable?  What do you think?